src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -55,6 +54,27 @@ protected AbstractSensitivityAnalysis(MatrixFactory matrixFactory) {
         this.matrixFactory = Objects.requireNonNull(matrixFactory);
     }
 
+    protected static Terminal getEquipmentRegulatingTerminal(Network network, String equipmentId) {
```

Comment:
@annetill Here I am trying to get the terminal that corresponds to the equipmentId, but I am not sure of the way to do it

Reply:
I have pushed a commit to help you (but the code could be improved if you want). Indeed it was what I have tried to avoid with another design of the factor, without any success from my colleagues :-)